src/preprocessor.ts

```diff
@@ -1,59 +1,29 @@
+import * as fs from 'fs-extra';
 import * as tsc from 'typescript';
-import * as nodepath from 'path';
-import { getTSConfig, getJestConfig } from './utils';
+import {getTSConfig} from './utils';
 // TODO: rework next to ES6 style imports
 const glob = require('glob-all');
-const getPackageRoot = require('jest-util').getPackageRoot;
-
-declare const global: any;
-
-const root = getPackageRoot();
-const jestConfig = getJestConfig(root);
-const { collectCoverage } = jestConfig;
-const {
-    testRegex,
-    coverageDirectory,
-    coverageReporters,
-    collectCoverageFrom,
-    testResultsProcessor
-} = jestConfig.config;
-
-//setting up cache to global object to resultprocessor consume
-if (testResultsProcessor) {
-    global.__ts_coverage__cache__ = {};
-    global.__ts_coverage__cache__.sourceCache = {};
-    global.__ts_coverage__cache__.coverageConfig = { collectCoverage, coverageDirectory, coverageReporters };
-    global.__ts_coverage__cache__.coverageCollectFiles =
-        collectCoverage &&
-            testResultsProcessor &&
-            collectCoverageFrom &&
-            collectCoverageFrom.length ?
-            glob.sync(collectCoverageFrom).map(x => nodepath.resolve(root, x)) : [];
-}
+const nodepath = require('path');
 
 export function process(src, path, config) {
     if (path.endsWith('.ts') || path.endsWith('.tsx')) {
         const transpiled = tsc.transpileModule(
             src,
             {
-                compilerOptions: getTSConfig(config.globals, collectCoverage),
+                compilerOptions: getTSConfig(config.globals, config.collectCoverage),
                 fileName: path
             });
 
         //store transpiled code contains source map into cache, except test cases
-        if (global.__ts_coverage__cache__) {
-            if (!testRegex || !path.match(testRegex)) {
-                global.__ts_coverage__cache__.sourceCache[path] = transpiled.outputText;
-            }
+        if (!config.testRegex || !path.match(config.testRegex)) {
+            fs.outputFileSync(nodepath.join(config.cacheDirectory, '/ts-jest/', new Buffer(path).toString('base64')), transpiled.outputText);
         }
 
         const start = transpiled.outputText.length > 12 ? transpiled.outputText.substr(1, 10) : '';
 
-        const modified = start === 'use strict'
+        return start === 'use strict'
             ? `'use strict';require('ts-jest').install();${transpiled.outputText}`
             : `require('ts-jest').install();${transpiled.outputText}`;
-
-        return modified;
     }
```

Comment:
Minor nit. It would be better to leave this part as is. The simpler return statements are, the more readable & less error prone they are, imo.